### DISCUSSION TITLE
Error: Returned error: VM Exception while processing transaction: revert

### DISCUSSION THREAD

**Discussion by ghost** [ConsenSys-archive/truffle#5696] [General Q&A]
can someone please help me...my tests are not running 
this is what my terminal is showing

 Contract: DecentralBank
    mock tether deployment
      √ should match the name and symbol correctly (1518ms)
    reward token deployment
      √ should matches the name and symbol correctly (635ms)
    decentralBank deployment
      √ should match the name successfully (291ms)
      √ contract has enough tokens (489ms)
    staking
      1) give tether tokens for staking

    Events emitted during test:
    ---------------------------

    Reward.Transfer(
      _from: <indexed> 0xF436B3716a5c3978cDB0051D8Fb5029c87493F7B (type: address),
      _to: <indexed> 0x723f97a6eCf4A4C5607153Adc8852A6B75442487 (type: address),
      _value: 1000000000000000000000000 (type: uint256)
    )

    Tether.transfer(
      _from: <indexed> 0xF436B3716a5c3978cDB0051D8Fb5029c87493F7B (type: address),
      _to: <indexed> 0x11C7bAbe0F3C8AFccE9b7b8cb1B59557F1E34752 (type: address),
      _value: 100000000000000000000 (type: uint256)
    )

    Tether.approval(
      _owner: <indexed> 0x11C7bAbe0F3C8AFccE9b7b8cb1B59557F1E34752 (type: address),
      _spender: <indexed> 0x723f97a6eCf4A4C5607153Adc8852A6B75442487 (type: address),
      _value: 100000000000000000000 (type: uint256)
    )

    ---------------------------

  4 passing (48s)
  1 failing

  1) Contract: DecentralBank
       staking
         give tether tokens for staking:
     Error: Returned error: VM Exception while processing transaction: revert

here is my repo link
https://github.com/Dania001/Dapp-Project

**Comment by eggplantzzz** [comment]
It looks like a transaction is reverting somewhere for some reason. Check your code and see if you can trace down which transaction is reverting.

**Comment by ghost** [comment]
hey @eggplantzzz i have checked everything soo many times i dont know what is wrong everything is compiling and migrating successfully i have no idea why its reverting back ....kindly help

**Comment by cliffoo** [comment]
Hey @Dania001 take a look at our [debugger docs](https://trufflesuite.com/docs/truffle/how-to/debug-test/use-the-truffle-debugger/) and try it out in your project, see if you can crack this issue a bit with it.

**Reply by ghost** [reply]
hey @cliffoo i have tried it did not help.........i dont know what to do please help

**Reply by lsqproduction** [reply]
Hi @Dania001 can you share your outputs from the debugging here?
